src/wizards/services/WizardService.ts

```diff
@@ -108,9 +111,25 @@ export class WizardService {
     }
   }
 
-  public submit(wizardManager: any, valid: boolean): void {
-    // eslint-disable-next-line no-console
-    console.log("submit", wizardManager, valid);
+  public async submit(wizardManager: WizardManager): Promise<void> {
+    let allStagesValid = false;
+    this.wizardsStates.forEach((wizardState) => {
+      allStagesValid = wizardState.stages.every(stage => stage.valid);
+    });
+
+    if (!allStagesValid) return;
+
+    const congratulatePopupModel: IAlertModel = {
```

Comment:
I would like that we avoid "deal" specific stuff in  the WizardService.  We should find another way to handle a successful click of the submit button (post-validation check).

Reply:
Agree.

But due to how the Wizard architecture is implemented, I don't see an "easy" or "quick" fix right now.
So:
1. Spent time thinking of making wizard "deal" unspecific
2. Still do 1. but later

@paweljanicki Do you see a quick win we can implement regarding "making wizard deal unspecific"?

Reply:
I think putting up the alert modal should be up to the stage, not the wizard.

Reply:
Yup I agree.
My question is when this should be refactored?

As I see it, it's the current way of the Wizard architecture

Reply:
[refac(submit): move congrats modal logic to submitStage](https://github.com/PrimeDAO/prime-deals-dapp/pull/227/commits/eca5c530dd3edb10647f03890b0bed27b2bd13f8)